src/navigator.js

```diff
@@ -405,10 +405,26 @@ $.extend( $.Navigator.prototype, $.EventSource.prototype, $.Viewer.prototype, /*
  * @function
  */
 function onCanvasClick( event ) {
-    if ( event.quick && this.viewer.viewport ) {
-        this.viewer.viewport.panTo(this.viewport.pointFromPixel(event.position));
-        this.viewer.viewport.applyConstraints();
+  if (event.quick && this.viewer.viewport && (this.panVertical || this.panHorizontal)) {
+    var target,
```

Comment:
This looks like it'll work, but it could be cleaner if you start with: 

```
var target = this.viewport.pointFromPixel(event.position);
```

...and then if `!this.panVertical` do: 

```
target.y = this.viewport.getCenter().y;
```

...and the same for the `!this.panHorizontal` case.

Otherwise looks great... thank you for fixing this!

Reply:
Thanks for the suggestion! I'm at my first JS project so every advice is more than welcome!
I'll change it tomorrow! It's a pleasure to contribute!

Thanks,
Francesco